packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -0,0 +1,202 @@
+  List<Source> get outputs => const <Source>[
+    Source.pattern('{BUILD_DIR}/main.dart.js'),
+  ];
+
+  @override
+  Future<void> build(Environment environment) async {
+    final String dart2jsOptimization = environment.defines[kDart2jsOptimization];
+    final BuildMode buildMode = getBuildModeForName(environment.defines[kBuildMode]);
+    final String specPath = fs.path.join(artifacts.getArtifactPath(Artifact.flutterWebSdk), 'libraries.json');
+    final String packageFile = FlutterProject.fromDirectory(environment.projectDir).hasBuilders
+      ? PackageMap.globalGeneratedPackagesPath
+      : PackageMap.globalPackagesPath;
+    final ProcessResult result = await processManager.run(<String>[
+      artifacts.getArtifactPath(Artifact.engineDartBinary),
+      artifacts.getArtifactPath(Artifact.dart2jsSnapshot),
+      '--libraries-spec=$specPath',
+      if (dart2jsOptimization != null)
+        '-$dart2jsOptimization'
+      else if (buildMode == BuildMode.profile)
+        '-O1'
+      else
+        '-O4',
+      '-o',
+      environment.buildDir.childFile('main.dart.js').path,
+      '--packages=$packageFile',
+      if (buildMode == BuildMode.profile)
+        '-Ddart.vm.profile=true'
+      else
+        '-Ddart.vm.product=true',
+      environment.buildDir.childFile('main.dart').path,
+    ]);
+    if (result.exitCode != 0) {
+      throw Exception(result.stdout + result.stderr);
+    }
+  }
+}
+
+/// Unpacks the dart2js compilation to a given output directory
+class WebReleaseBundle extends Target {
+  const WebReleaseBundle();
+
+  @override
+  String get name => 'web_release_bundle';
+
+  @override
+  List<Target> get dependencies => const <Target>[
+    Dart2JSTarget(),
+  ];
+
+  @override
+  List<Source> get inputs => const <Source>[
+    Source.pattern('{BUILD_DIR}/main.dart.js'),
+    Source.pattern('{FLUTTER_ROOT}/packages/flutter_tools/lib/src/build_system/targets/web.dart'),
+    Source.behavior(AssetOutputBehavior('assets')),
+    Source.pattern('{PROJECT_DIR}/web/index.html'),
+  ];
+
+  @override
+  List<Source> get outputs => const <Source>[
+    Source.pattern('{OUTPUT_DIR}/main.dart.js'),
+    Source.pattern('{OUTPUT_DIR}/assets/AssetManifest.json'),
+    Source.pattern('{OUTPUT_DIR}/assets/FontManifest.json'),
+    Source.pattern('{OUTPUT_DIR}/assets/LICENSE'),
+    Source.pattern('{OUTPUT_DIR}/index.html'),
+    Source.behavior(AssetOutputBehavior('assets'))
+  ];
+
+  @override
+  Future<void> build(Environment environment) async {
+    for (File outputFile in environment.buildDir.listSync(recursive: true).whereType<File>()) {
+      if (!fs.path.basename(outputFile.path).contains('main.dart.js')) {
+        continue;
+      }
+      outputFile.copySync(
+        environment.outputDir.childFile(fs.path.basename(outputFile.path)).path
+      );
+    }
+    environment.projectDir
+      .childDirectory('web')
+      .childFile('index.html').copySync(fs.path.join(environment.outputDir.path, 'index.html'));
```

Comment:
nit: unless this is dartfmt'ed I'd move `copySync` to the next line too.

Reply:
Fixed